subprojects/build-cache/src/main/java/org/gradle/caching/internal/controller/NextGenBuildCacheController.java

```diff
@@ -31,20 +32,26 @@
 import org.apache.commons.io.input.UnsynchronizedByteArrayInputStream;
 import org.apache.commons.io.output.NullOutputStream;
 import org.apache.commons.io.output.TeeOutputStream;
+import org.gradle.api.internal.cache.StringInterner;
 import org.gradle.caching.BuildCacheEntryWriter;
 import org.gradle.caching.BuildCacheKey;
 import org.gradle.caching.internal.CacheableEntity;
 import org.gradle.caching.internal.DefaultBuildCacheKey;
 import org.gradle.caching.internal.controller.CacheManifest.ManifestEntry;
 import org.gradle.caching.internal.controller.service.BuildCacheLoadResult;
 import org.gradle.caching.internal.origin.OriginMetadata;
+import org.gradle.caching.internal.packaging.impl.RelativePathParser;
```

Comment:
❓ Shall we create a copy of this parser, so we can throw out the whole `packaging` package when we switch over completely?

Reply:
We can do that. The original parser was anyway annoying since it required folders to end with '/'. Moved new class as static inner class of NextGenBuildCacheController

Reply:
On second thought, we also have tests for RelativePathParser, so we would need to duplicate that. I will revert that change and use `org.gradle.caching.internal.packaging.impl.RelativePathParser`. I think it would be good we move it to other package once we remove that package.

Reply:
OK, your call.

Reply:
Please add any follow-up ideas as TODOs in the code so we don't forget.